pkg/asset/agent/image/baseiso.go

```diff
@@ -117,7 +111,12 @@ func (i *BaseIso) Generate(dependencies asset.Parents) error {
 
 	var baseIsoFileName string
 	var err error
+	//agentManifests.ClusterImageSet = nil
 	if agentManifests.ClusterImageSet != nil {
+		// Override rhcosArch = x86_64 if InfraEnv has Arch defined
+		if agentManifests.InfraEnv.Spec.CpuArchitecture != "" {
+			rhcosArch = agentManifests.InfraEnv.Spec.CpuArchitecture
```

Comment:
Let's define a local variable that defaults to "x86_64". I'm not comfortable with changing a global here, especially when the name is shadowed by local variables in lots of functions.